### ISSUE TITLE
There's no way for "Enter-PSSesion -Session" to block the calling process until user types "exit"

### ISSUE THREAD

**Issue by wm2015email** [PowerShell/PowerShell#16292]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

    $s = New-PSSession -computer "remotecomputer1"  
    Enter-PSSesion -Session $s    #<<-Why not at a new powershell flag  here called "-wait"??
    # I assume at this point there are two thread running...
    # one thread for the main script, and the other for PSSession
    #  Wait-PSSession -Session $s <== THE MISSING COMMAND THAT IS NONE EXISTENT IN POWERSHELL
    # Now the main thread falls though and doesn't wait!!!!! because there's no command to stop this????
    write-host "Shouldn't get here until user types "exit" in session on "remotecomputer1"
    exit

If this isn't a bug, then it needs to have an example of how to "Wait-PSSession  -Session" is implemented in the PowerShell documention for "Enter-PSSession -Session" command.

Why this matters?  Because what happens if I call this from WSL:

    $ powershell.exe -nologo -File MyRemoting.ps1

It kills the interactive session because it doesn't block the main thread until the interactive session remoting session finishes.

I can get around this by doing

    $ powershell.exe -nologo -noexit -File MyRemoting.ps1

But, then i need to type exit everytime to get out of powershell also!  Then, I want to throw away all my windows computers and install linux instead.

### Expected behavior

```console
A command for block the execution of "Enter-PSSession -Session".. but it doesn't exist!
```

### Actual behavior

```console
Enter-PSSession falls through as non-blocking and powershell keeps running the main powershell thread... (Bad/Wrong behavior)
```

### Error details

```console
None...
```

### Environment data

```powershell
None.
```

### Visuals

None.

**Comment by mrmeszaros** [comment]
@PaulHigin I have a pretty similar use case, where I want to acquire an interactive session in the middle of a script.
What other means exist to achieve that, beside `Enter-PSSession`?

In detail I have a utility script that sets up a remote session, and I would like to have the tear-down part after it. Like

```powershell
# Set up parts
# ...
# Enter the interactive session, command execution is yielded to the user (me)
Enter-PSSession $pss
# Enter-PSSession does not return, until exit/Exit-PSSession is called
# Tear down that I have to do manually currently
# finally exit pwsh
exit
```

I suggest, that if it is unfeasible to implement a blocking way to call `Enter-PSSession` (with a flag like `-Wait` or `-Block` or `-Interactive`), it could put up a warning that calling the command from a script will not result in what a user could expect.

**Comment by PaulHigin** [comment]
Why do you want to convert a running script into an interactive session?  You can have PowerShell prompt for input, if that is needed.  But currently the only way for a running script to allow interaction is through the script debugger.

**Comment by mrmeszaros** [comment]
I have a pretty special use case.
I want to connect to windows machines from a linux machine in a single step. There are many machines, so I have populated my `/etc/hosts` file for shell autocompletion purposes.

```fish
function pwsh-remote --argument host --description 'Open Powershell to remote...'
  pwsh -NoExit -Command "
  \$creds = Get-Credential -UserName the-doctor
  \$pss = New-PSSession -ComputerName $host -Authentication Negotiate -Credential \$creds
  Enter-PSSession \$pss
  # This should run after exiting the remote and exit the main pwsh
  exit
  "
end
complete -c pwsh-remote -x -d "Remote" -a "(__fish_complete_user_at_hosts)"
```

I have tried using `Register-EngineEvent` with `-SourceIdentifier PowerShell.Exiting` (and `-Forward` on the remote) hoping to catch the event of exiting the remote PSSession but with no avail.

Basically, I would like to have callbacks on exiting the remote session.
Now I know this is an issue tracker and no Q/A forum so should it bother You, please nudge me towards the right forum/person?

(sadly [my stackoverflow question](https://stackoverflow.com/questions/69782883/how-to-forward-powershell-exiting-event-when-remoting-on-linux) did not provide a satisfying answer yet)

**Comment by kasini3000** [comment]
hi mrm:
[kasini3000](https://gitee.com/chuanjiao10/kasini3000)    
win,linux devops automation batch script framework.(It is similar to ansible) 
Open source, free, cross-platform 

English manual: https://gitee.com/chuanjiao10/kasini3000/blob/master/docs/kasini3000_manual.xlsx
script dependencies 【c:\ProgramData\kasini3000\nodelist.csv】

1 Multi-process batch execution. k-commit.ps1
2 Multi-threaded batch execution. k-commit-rs.ps1 #need powershell module poshrsjob
setup:
copy files from
c:\ProgramData\kasini3000\lib\kasini3000\ProgramFiles_WindowsPowerShell_Modules
to
$psmodulepath

----
c:\ProgramData\kasini3000\k_run_ip_port_l2w_user_pwd.ps1
or 
/etc/kasini3000/k_run_ip_port_l2w_user_pwd.ps1 #linux 
this script not dependencies 【c:\ProgramData\kasini3000\nodelist.csv】,but need parameter username ,password.

**From linux master to win node, no support is provided.**
